specs/ExecuteScriptWithResult.md

```diff
@@ -0,0 +1,268 @@
+# Background
+Since the current ExecuteStrip interface is a little shabby, it is necessary
+to provide a new interface to let the user get more infomation, and easier to use.
+The new interface will provide the exception infomation if the script execute
+failed, and provide a new method to try to get the string to resolve the problem
+that the old interface is not friendly to the string return type.
+
+In this document we describe the updated API. We'd appreciate your feedback.
+
+# Description
+We propose extending `CoreWebView2` to provide an `ExecuteScriptWithResult` 
+method. The method will return a struct to manage the execute result, which can
+get the raw result and string if execute success, and can get exception when
+execute failed.
+
+# Examples
+The following code snippets demonstrate how the ExecuteScriptWithResult can be used:
+## Win32 C++
+``` cpp
+void ScriptComponent::ExecuteScriptWithResult(LPCWSTR script)
```

Comment:
The sample code should demonstrate something that an end developer would actually want to implement. The below has much MessageBox and Debug.WriteLine rather than trying to do something with the result. Perhaps you can take the existing ExecuteScript code as an example but modified to handle exceptions